app/controllers/products_controller.rb

```diff
@@ -13,20 +15,31 @@ def show
     end
 
     def destroy
-        @product = Product.find(params[:id])
-        @product.destroy!
+        begin
+            @product = Product.find(params[:id])
+            authorize @product, :destroy?
+            @product.destroy!
 
-        redirect_to controller: "home",action: "index"
+            redirect_to controller: "home",action: "index"
+        rescue
+            redirect_to controller: "home",action: "index"
+        end
     end
 
     def create
-        # fileName = params[:article][:image].original_filename
-        # params[:article][:image] = fileName
         @product = Product.new(product_params)
-        if @product.save
-            redirect_to controller: 'home', action: 'index'
-        else
-            render 'new'
+        begin
+            authorize @product, :create?
+            if @product.save
+                redirect_to controller: 'home', action: 'index'
+            else
+                render 'new'
+            end
+            flash[:notice] = "New Product added"
+        rescue Exception => error
```

Comment:
Instead of handling exception in each controller.

You can write something in ApplicationController.


```class ApplicationController < ActionController::API
  include Pundit

  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized

  private

  def user_not_authorized
    render json: { error: 'Unauthorized for the action !' }, status: 401
  end
end